pkg/webhooks/policymutation.go

```diff
@@ -145,7 +145,9 @@ func generatePodControllerRule(policy kyverno.ClusterPolicy, log logr.Logger) (p
 
 	// scenario A
 	if !ok {
-		controllers = "all"
+		if controllers == "all" {
+			controllers = "DaemonSet,Deployment,Job,StatefulSet"
+		}
```

Comment:
What if controllers != "all"? It will pass in "" to `generateRulePatches()` in line 165.